internal/infrastructure/middleware/validate_api.go

```diff
@@ -36,7 +36,7 @@ func (x xrhiAlwaysTrue) ValidateXRhIdentity(xrhi *identity.XRHID) error {
 func InitOpenAPIFormats() {
 	// TODO Review all the regular expressions
 	openapi3.DefineStringFormat("url", `^https?:\/\/.*$`)
-	openapi3.DefineStringFormat("realm", `^(([A-Z0-9][A-Z0-9\-]*[A-Z0-9])|[A-Z0-9]+\.)*([A-Z]+|xn\-\-[A-Z0-9]+)\.?$`)
+	openapi3.DefineStringFormat("realm", `^([A-Z]|[A-Z][A-Z0-9\-]*[A-Z0-9])(\.([A-Z]|[A-Z][A-Z0-9\-]*[A-Z0-9]))+$`)
```

Comment:
@tiran can you please eyeball this?  Hostnames with leading digits are permitted by https://datatracker.ietf.org/doc/html/rfc1123#section-2 (e.g. https://9292.nl/) and I think we should not prohibit it in our API.

Reply:
Ah nevermind, this is for the REALM name.  I should read more closely.

Reply:
```
Realm           ::= KerberosString
KerberosString  ::= GeneralString (IA5String)
```

So... I think we should either allow the leading digits in the hostname, or else relax this check closer to the RFC spec, i.e. that it is IA5String (printable ASCII, essentially).  This is notwithstanding the *convention* that realm corresponds to an upper-cased DNS name.  @tiran @cryptomilk what do you think?

Reply:
Follow up on HMS-3429